research-software-directory/RSD-as-a-service
# Improve e2e tests stability

# Improve e2e test stability and fix shared session

Changes proposed in this pull request:
* e2e tests extended with concurrent login test, including app fix concerning found error
* e2e tests concerning external api calls to doi.org, crossref and datacite are stubbed and return locally saved json data. It seems that these calls often timeout when executed in CI.
* e2e "project" created to save all DOI api calls that need to be performed. This project should be runned if DOI list used in tests is changed.   
* upgrade playwright to latest version v 1.32.2
* upgrade e2e dockerfile to use new playwright image v1.32.0
* comment out SURF and ORCID login options in e2e.env because these options are not available in CI (missing secrets)

How to test:
* `make e2e-tests` to run all tests locally. The test should pass
* e2e tests are runnen in CI on each PR. So e2e tests of this PR should be also passing

PR Checklist:
